src/portal/mod.rs

```diff
@@ -104,48 +107,65 @@ impl Keyring {
             }
         };
 
-        let key = keyring.derive_key(secret);
+        //let key = Lazy::new(|| keyring.derive_key(secret));
 
         Ok(Self {
             keyring: Mutex::new(keyring),
             path: path.as_ref().to_path_buf(),
             mtime: Mutex::new(mtime),
-            key,
+            secret: Zeroizing::new(secret.to_vec()),
+            key: OnceCell::new(),
         })
     }
 
+    /// Retrieve the number of items
+    ///
+    /// This function does not require for a key to be derived.
+    pub async fn num_items(&self) -> usize {
```

Comment:
Would name it `n_items`

Reply:
Should the dbus side of things have a similar method? if so, please add it and add it to wrapper keyring (src/keyring)